### PR TITLE
chore: only inject when html is going to be rendered

### LINKED ISSUE #24297: Cypress began to inject additional markup into the downloaded csv, tsv & txt files (after v. 10.9.0)
### Current behavior

Downloaded or opened in the new tab csv, tsv or txt file contains a very long markup code before the actual content of the file, for example 
```
<head> <script type='text/javascript'> document.domain = 'mydomain.us'; const cypressConfig = { modifyObstructiveThirdPartyCode: false, modifyObstructiveCode: true }; !function(a){var o={};function e(i){if(o[i])return o......(178000 more characters).....,0:function(a,o,e){a.exports=e("./injection/cross-origin.js")}}); </script> </head>
```
|column_name |table_name | value_type | encoding_id | ordaunits | notes |
|:-------------------|:-------------|:-----:|:-----:|:-----:|:--------------------------------------------------------------|
|ins_index|deatch|11|0|1|A numerical index used to distinguish the cases where we have multiple death records for a participant.|
|date_of_death|death|51|0|4|Date of death|
|dsource|death|41|1970|2|The data source is either E/W for England & Wales or SCOT for Scotland.|
|eid|death|11|0|0|This identifier is the same encoded id used in the main dataset.|
|source|death|11|261|3|The particular data format in which this record was received.|

### Desired behavior

Downloaded file should not contain any <head><script></script><head> markup.

It should contain actual csv/tsv/txt content, for example:
```
column_name	table_name	value_type	encoding_id	orda	units	notes
ins_index	deatch	11	0	1		A numerical index used to distinguish the cases where we have multiple death records for a participant.
date_of_death	death	51	0	4		Date of death
dsource	death	41	1970	2		The data source is either E/W for England & Wales or SCOT for Scotland.
eid	death	11	0	0		This identifier is the same encoded id used in the main dataset.
source	death	11	261	3		The particular data format in which this record was received.
```
The issue started to happen after we updated Cypress from 10.8.0 to 10.9.0. Also occurs in Cypress 10.10.0

### Test code to reproduce

```
cy.readFile("downloads/record_column.csv").then((text) => {
      cy.readFile("./fixtures/bencmarkFiles/record_column.csv").should("eq", text);
});
```

### Cypress Version

^10.8.0

### Node version

16.13.1

### Operating System

macOS 12.5.1

### Debug Logs

_No response_

### Other

Sadly, the issue is not reproducible when we're trying to download sample CSVs from here: https://www.sample-videos.com/download-sample-csv.php , it happens only on our product, which is closed to the public :(

### PR DESCRIPTION
**Pull request by AtofStryker**
and if a content-type exists, make sure it contains html (which is valid for xhtml and other mime types)

- Closes #24297 

### User facing changelog
Fixes an issue where Cypress would accidentally inject into `csv`s and other non-html mimetypes. 

### Additional details
Cypress was injecting `fullCrossOrigin` and sometimes `full` into `csv`s and other mime types when it shouldn't. This check adds an additional check to renderedHTML that 
### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? N/A
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by chrisbreiding — APPROVED** [review]

File: packages/proxy/lib/http/response-middleware.ts [hunk 1]
```diff
@@ -93,7 +97,7 @@ function resContentTypeIsJavaScript (res: IncomingMessage) {
   )
 }
 
-function isHtml (res: IncomingMessage) {
+function isNotJavascript (res: IncomingMessage) {
   return !resContentTypeIsJavaScript(res)
 }
```
Comment (flotwig) [inline packages/proxy/lib/http/response-middleware.ts]: Nice cleanup. You could go just a little bit further and make this less confusing, delete this function and update the `isNotJavascript`s to  `isJavascript: resContentTypeIsJavaScript(res)`. The less inversion and indirection the easier it is to understand.

**Review by flotwig — COMMENTED** [review]
Related issue about XHTML, not fixed by this PR: #15962

File: packages/proxy/lib/http/response-middleware.ts (hunk 1 above)
Comment (AtofStryker) [reply packages/proxy/lib/http/response-middleware.ts]: Thank you! And agreed. Added that in dcccf74da3295d68621f219bcb6d5b0d7feef880.

Comment (flotwig) [reply packages/proxy/lib/http/response-middleware.ts]: I was thinking that the `isNotJavascript`s could be updated to `isJavascript`, to minimize inversion, but this is an improvement too.

**Review by flotwig — APPROVED** [review]

**Pull request merged**